### PR TITLE
Implement Args builder for getObject API

### PR DESCRIPTION
**Pull request by anjalshireesh**
New Argument class: `GetObjectArgs`

Arguments supported:
 - offset (Long)
 - length (Long)
 - ssec (ServerSideEncryptionWithCustomerKey)

Since there are multiple APIs / args classes that need `ssec` as an argument, introduced an intermediate abstract class `SsecObjectArgs` that supports the `ssec` argument, and provides common logic related to it e.g. validations.

Since the return type of the methods that support downloading the object as a file is `void`, we cannot name it the same as `getObject`. So introduced a new method called `downloadObject` for downloading the object, which internally uses `getObject`. And introduced a new args class `DownloadObjectArgs` for it.

Also,
  - Enhanced the method `FunctionalTest#handleException` to support  logging of the arguments as well.
  - Moved the class `ServerSideEncryptionWithCustomerKey` outside of `ServerSideEncryption` and made it into a standalone class with the name `ServerSideEncryptionCustomerKey`

### CODE REVIEW

File: api/src/main/java/io/minio/MinioClient.java [hunk 1]
```diff
@@ -1626,7 +1626,7 @@ public ObjectStat statObject(String bucketName, String objectName)
    *
    * <pre>Example:{@code
    * ObjectStat objectStat =
-   *     minioClient.statObject("my-bucketname", "my-objectname", ssec);
+   *     minioClient.statObject("my-bucketname", "my-objectname", sse);
```
Comment (balamurugana) [inline api/src/main/java/io/minio/MinioClient.java:1629]: This is still `ssec`. You could revert this change.

File: api/src/main/java/io/minio/MinioClient.java [hunk 2]
```diff
@@ -1713,7 +1714,7 @@ public ObjectStat statObject(StatObjectArgs args)
           InternalException, InvalidBucketNameException, InvalidKeyException,
           InvalidResponseException, IOException, NoSuchAlgorithmException, XmlParserException {
     checkArgs(args);
-    checkReadRequestSse(args.ssec());
+    args.validateSsec(baseUrl);
```
Comment (balamurugana) [inline api/src/main/java/io/minio/MinioClient.java:1717]: Is it a good idea to have this check inside `checkArgs()`?

File: api/src/main/java/io/minio/SsecObjectArgs.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+ */
+
+package io.minio;
+
+import okhttp3.HttpUrl;
+
+public abstract class SsecObjectArgs extends ObjectArgs {
+  protected ServerSideEncryptionCustomerKey ssec;
+  public static final int TYPE_READ = 0;
+  public static final int TYPE_WRITE = 1;
```
Comment (balamurugana) [inline api/src/main/java/io/minio/SsecObjectArgs.java:24]: Are these constants still needed?

File: api/src/main/java/io/minio/MinioClient.java (hunk 1 above)
Comment (anjalshireesh) [reply api/src/main/java/io/minio/MinioClient.java:1629]: Will revert

File: api/src/main/java/io/minio/MinioClient.java (hunk 2 above)
Comment (anjalshireesh) [reply api/src/main/java/io/minio/MinioClient.java:1717]: `checkArgs` accepts the base class `BaseArgs`, so the `validateSsec()` method which is defined in `SsecObjectArgs` won't be available there.

File: api/src/main/java/io/minio/SsecObjectArgs.java (hunk 3 above)
Comment (anjalshireesh) [reply api/src/main/java/io/minio/SsecObjectArgs.java:24]: Not required! Thanks, will remove.
